### ISSUE TITLE
Write-Output and Write-Host do not enumerate enumerators, notably not $input

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#18826]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Generally, using `$foo` (implicit output) and `Write-Output $foo` exhibits the same behavior, but in the case of the automatic `$input` variable, which is an _enumerator_ (rather than an _enumerable_) that isn't the case:

Note: The same applies analogously to `Write-Host`, which simply prints the enumerator's _type name_; the examples below focus on `Write-Output`.

```powershell
1..3 | & { Write-Output $input } 

# On Windows; use `cat` on Unix
1..3 | & { Write-Output $input | findstr.exe ^ } 
```

Obviously, there' s no good reason to use `Write-Output` here, but it should still work.

The for-display / stringified output that results is the implicit `Format-Table` formatting of the enumerator's only property, `Current`, which - in the absence of actually starting the enumeration with `.MoveNext()` - is _always `$null`_ - irrespective of how many objects the enumerator represents -  and therefore results in a blank column value.

In short: the for-display output and the stringified output for external programs that it is based on is useless, and both `Write-Output` and `Write-Host` should perform enumeration when given an enumerator.
Conceivably, there are additional contexts where an analogous change is called for.

### Expected behavior

```console
1
2
3
1
2
3
```

### Actual behavior

```console
Current
-------



Current
-------
```

### Environment data

```powershell
PowerShell Core 7.3.1
```

**Comment by 237dmitry** [comment]
Looked in 5.1 -- the same.
Works if:
```
1..3 | & { (Write-Output $input) }
1..3 | & { (Write-Output $input) | grep ^ }
```

**Comment by mklement0** [comment]
Yes, `(...)` in a pipeline context forces enumeration, so you can use that as a _workaround_ - though, as noted in the OP, you can simply _omit_ `Write-Output` - which isn't just a workaround, but the preferred solution.

The point of this issue is the unhelpful behavior of `Write-Output` and `Write-Host` with enumerators.

**Comment by 237dmitry** [comment]
> so you can use that as a workaround 

As you noted `$input` works without workaround. It's strange to use this construction. I would rather use `$input | Write-Host`.

**Comment by mklement0** [comment]
My previous comment states what this issue is about in (logically) _positive_ terms. Let me complement with a _negative_ definition: this issue _isn't_ about what you would rather use or whether there are _other_ solutions.

**Comment by 237dmitry** [comment]
> this issue isn't about what you would rather use or whether there are other solutions.

I did not intend to offer you anything... Just showed under what conditions it works and clarified that this problem is a hundred years old and everyone thought that it should be so.

**Comment by mklement0** [comment]
> Just showed under what conditions it works

The initial post already showed the preferable solution, so there was no need for that.

> this problem is a hundred years old

How old a problem is should have no bearing on whether it deserves fixing.

> everyone thought that it should be so.

That is both (a) entirely speculative and (b) strikes me as extremely unlikely, given the symptom shown in the initial post - if anyone can come up with a good reason for this behavior, I'm all ears.

In short: 

* That something hasn't been reported before and/or that not many users have run into a given problem should not factor into whether the problem should be fixed.

**Comment by SeeminglyScience** [comment]
The Engine WG discussed this and agree it's a bug that should be fixed.
